net/dns/resolver/tsdns.go

```diff
@@ -62,6 +63,24 @@ type packet struct {
 	addr netaddr.IPPort // src for a request, dst for a response
 }
 
+// ResolverName is the configuration for a DNS resolver.
+type ResolverName struct {
+	// Addr is the address of the DNS resolver, one of:
+	//  - A plain IP address for a "classic" UDP+TCP DNS resolver
+	//  - An IP:port pair
+	//  - [TODO] "tls://resolver.com" for DNS over TCP+TLS
+	//  - "http(s?)://resolver.com/query-tmpl" for DNS over HTTP(S?)
+	Addr string
+
+	// BootstrapResolution is an optional suggested resolution for the
+	// DoT/DoH resolver, if the resolver URL does not reference an IP
+	// address directly.
+	// BootstrapResolution may be empty, in which case clients should
+	// look up the DoT/DoH server using their local "classic" DNS
+	// resolver.
+	BootstrapResolution []netaddr.IP
```

Comment:
could these be URL parameters on Addr if Addr is a URL-ish thing? Maybe not for efficiency. Is this used yet?

Reply:
I don't want to deviate from the tailcfg type by inventing a new type here. Otherwise, sounds possible.